### ISSUE TITLE
Digitizing: Cut/Paste behaviour counter intuitive (to me)

### ISSUE THREAD

**Issue by rduivenvoorde** [qgis/QGIS#43267]
Context: small table with some point features in postgres (being position of monitoring vehicles). I hoped it would be possible to copy/paste a vehicle from one side of the country to another.

Observations:
- when you cut a feature, pan/zoom to another part of the country, pasting a feature put's it back on the OLD position, not (what I was thinking) somewhere in the middle of current mapcanvas extent
- cutting a feature, seems to delete/clean up the id/pk field (I think it should NOT, you cut a feature to be able to PASTE it back, isn't it?) 

To test, in a postgis db create a spatial table:

```
CREATE TABLE public.cars (
	id int4 NOT NULL,
	description varchar NOT NULL,
	projectid int4 NULL,
	geom geometry(POINT, 4326) NULL,
	CONSTRAINT enforce_dims_geom CHECK ((st_ndims(geom) = 2)),
	CONSTRAINT enforce_geotype_geom CHECK (((geometrytype(geom) = 'POINT'::text) OR (geom IS NULL))),
	CONSTRAINT enforce_srid_geom CHECK ((st_srid(geom) = 4326)),
	CONSTRAINT cars_pk PRIMARY KEY (id)
);
```

add some features with QGIS and try to copy/paste them to another part of the country. Observe that  if you paste it the id is gone (and you are being asked to add a new one).

The first item: that the primary key is removed seems a bug to me?

The second item: that pasting does not follow the cursor or mapcanvas is maybe more a feature request? This could be implemented, either as: paste it in the middle of the mapcanvas (so NOT on the exact old position), or maybe even better: add a mapcanvas context menu with 'paste feature here' in case there is a copied/cut feature in memory?

**Comment by MrChebur** [comment]
@rduivenvoorde This is not a bug. The Primary key must be unique, so it is new every time any object is added.

If you implement your second suggestion, then when you copy data from one layer to another, you will need to re-do the georeferencing of objects.

Use "move feature" tool to move objects in one layer.

**Comment by rduivenvoorde** [comment]
@MrChebur yes, you are probably right... I copy/paste within one layer, so indeed I MOVE objects... But is the 'cut'-action not also meaned to be a 'move' action?

To be more precise:
- if you paste an earlier cut feature in a different layer: cleanup the PK
- if you paste an earlier cut feature in the same layer: paste it with old PK (and now cleanup PK in the cut feature (because the user could optionally paste it again)?

Not sure what you mean with 'georeferencing'...  you mean reprojecting in case of different crs's? Or do I miss something?

**Comment by MrChebur** [comment]
@rduivenvoorde What if you want to copy an object to another layer at the same location as the original layer? The implementation of the functionality you suggest won't allow you to do that (unless you make a separate tool for it).

**Comment by rduivenvoorde** [comment]
@MrChebur no, not overwriting the current behavior. I think the context menu would be an addition, so you can still 'paste' it on exact the same location (in same or other layer), but if you move your mouse and use right click there would be a context menu asking 'Paste here'

**Comment by lbartoletti** [comment]
> @MrChebur no, not overwriting the current behavior. I think the context menu would be an addition, so you can still 'paste' it on exact the same location (in same or other layer), but if you move your mouse and use right click there would be a context menu asking 'Paste here'

@rduivenvoorde the right click would be a kind of "copy/move" tool
